lambdas/service/app.py

```diff
@@ -655,204 +655,270 @@ def repository_search(entity_type: str, item_id: str):
         raise NotFoundError(msg=e)
 
 
-@app.route('/repository/files', methods=['HEAD', 'GET'], cors=True)
-@app.route('/repository/files/{file_id}', methods=['GET'], cors=True)
+generic_object_spec = schema.object(additional_properties=True)
+array_of_object_spec = schema.array(generic_object_spec)
+hit_spec = schema.object(
+    additional_properties=True,
+    protocols=array_of_object_spec,
+    entryId=str,
+    samples=array_of_object_spec,
+    specimens=array_of_object_spec,
+    cellLines=array_of_object_spec,
+    donorOrganisms=array_of_object_spec,
+    organoids=schema.array(str),
+    cellSuspensions=array_of_object_spec,
+)
+
+page_spec = schema.object(
+    hits=schema.array(hit_spec),
+    pagination=generic_object_spec,
+    termFacets=generic_object_spec
+)
+
+
+def filters_param_spec(facets):
+    array_pair_spec = schema.array({}, minItems=2, maxItems=2)
+    return params.query(
+        'filters',
+        schema.optional(
+            application_json(
+                # Swagger UI doesn't produce a useful example for this schema,
+                # so several are manually provided below.
+                schema.object(**{
+                    facet: schema.optional({
+                        'oneOf': [
+                            schema.object(**{'is': schema.array({})}),
+                            *[
+                                schema.object(**{rel: array_pair_spec})
+                                for rel in ['contains', 'within', 'intersects']
+                            ]
+                        ]
+                    })
+                    for facet in facets
+                })
+            )
+        ),
+        description=format_description('''
+            Criteria to filter entities from the search results.
+
+            Each filter consists of a facet, a relationship, and an array of
+            relationship parameters. The available relationships are "is",
+            "within", "contains", and "intersects". See the `sort` parameter for
+            a list of filter-able facets.
+
+            For the "is" relationship, the parameters are simple values and
+            multiple parameters are combined using "or" logic. For example,
+            `{"fileFormat": {"is": ["fastq", "fastq.gz"]}}` will select entities
+            where the file format is either "fastq" or "fastq.gz". For the other
+            relationships, the parameters are pairs specifying upper and lower
+            bounds, and multiple parameter pairs are combined using "and" logic.
+            For example, `{"donorCount": {"within": [[1,5], [5,10]]}}` will
+            select entities whose donor organism count falls within both ranges,
+            i.e., is exactly 5.
+
+        '''),
+        # apidev.py claims that this parameter should not be provided when using
```

Comment:
This is a bit hand-waving. If this is indeed a claim made by `apidev.py`, could you file an issue and turn this into a FIXME comment? Otherwise, could you please point to the component that actually makes this claim?

Reply:
From the localhost Swagger UI:

```
Structural error at paths./repository/files.get.parameters.0
should not have `style`, `explode`, `allowReserved`, `example`, or `examples` when `content` is present
Jump to line 483
```
Repeated for every route that uses the filters param.

Reply:
Perhaps I should have been more clear that the error comes from within Swagger, not the code in our script.

Reply:
Yes. If a library used by a scripts rejects an input, the rejection should be attributed to the library, not the script, or at least not only the script.

Reply:
To fix the validation error, the example must be nested underneath `application/json` as [documented](https://swagger.io/docs/specification/adding-examples/).

Reply:
Oh and this might lead to a reversion of the reversion of the reversion of that type hint.